### PR TITLE
Perfect square 1 29 21

### PR DESCRIPTION
**Pull request by ParkerLBrown**
Function to find if a given int is a perfect square

### CODE REVIEW

**Review by dave22000 — CHANGES_REQUESTED** [review]
@ParkerLBrown  can you add comment with link to the codewars challenge as the first line of each js function

**Comment by ParkerLBrown** [comment]
@dave22000 just submitted with links

File: CodingWars Algos/Perfect Square/Perfect_Square.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
+// https://www.codewars.com/kata/54c27a33fb7da0db0100040e/train/javascript
+
+function isPerfectSquare (num) {
+  var i = 0;
+  while (i <= num) {
```
Comment (dave22000) [inline CodingWars Algos/Perfect Square/Perfect_Square.js]: @ParkerLBrown this is functionally correct but there are efficiency gains and cleanup that can be done:

you can simplify this logic - once i*i > num, you're done.

`for (var i = 0; i * i <= num; i++)`

Also, you don't need the (i * i !== num) conditional check at all - if (i * i === num) return true is the only part you need

Alternatively, if you want to get fancy, https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/sqrt

Comment (ParkerLBrown) [reply CodingWars Algos/Perfect Square/Perfect_Square.js]: @dave22000  function was refactored for efficiency, great suggestions, thanks! new function looks a lot prettier and more concise

**Pull request merged**